### PR TITLE
[mempool] 1. dedicated IO_POOL 2. get seq_num with shared state view

### PR DESCRIPTION
**Pull request by msmouse**
### Description

see individual commits

### Test Plan

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2428)

### CODE REVIEW

**Review by lightmark — APPROVED** [review]
nice hair wash!

**Review by zekun000 — COMMENTED** [review]
why not a dedicated pool for storage instead of mempool?

**Comment by lightmark** [comment]
> why not a dedicated pool for storage instead of mempool?

we will do that too.

**Comment by msmouse** [comment]
@zekun000  mempool shouldn't be doing IO in the global pool anyway

**Review by grao1991 — APPROVED** [review]

File: mempool/src/thread_pool.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use once_cell::sync::Lazy;
+
+pub(crate) static IO_POOL: Lazy<rayon::ThreadPool> = Lazy::new(|| {
```
Comment (sitalkedia) [inline mempool/src/thread_pool.rs:8]: Will having a static global threadpool cause any issue for unit tests that run in parallel?

Comment (msmouse) [reply mempool/src/thread_pool.rs:8]: I don't see why it would.

Comment (sitalkedia) [reply mempool/src/thread_pool.rs:8]: Hmm.. the concern is one bad test hogging the IO pool can make other tests flaky as well I guess? Is there any reason this needs to be static global? Can we not make it one thread pool per mempool?

Comment (msmouse) [reply mempool/src/thread_pool.rs:8]: Most rayon calls in the repo are using the global pool anyway?

**Pull request merged**